plugins/modules/mso_backup_schedule.py

```diff
@@ -0,0 +1,194 @@
+    username: admin
+    password: SomeSecretPassword
+    state: query
+  delegate_to: localhost
+
+- name: Set backup schedule
+  cisco.mso.mso_backup_schedule:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    interval_unit: hours
+    interval_length: 7
+    remote_location: ansible_test
+    state: present
+  delegate_to: localhost
+
+- name: Set backup schedule with date and time
+  cisco.mso.mso_backup_schedule:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    interval_unit: days
+    interval_length: 1
+    remote_location: ansible_test
+    remote_path: test
+    start_time: 20:57:36
+    start_date: 2023-04-09
+    state: present
+  delegate_to: localhost
+
+- name: Delete backup schedule
+  cisco.mso.mso_backup_schedule:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    state: absent
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from datetime import date, datetime, time
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        start_date=dict(type='str'),
+        start_time=dict(type='str'),
+        interval_unit=dict(type='str', choices=['hours', 'days']),
+        interval_length=dict(type='int'),
+        remote_location=dict(type='str'),
+        remote_path=dict(type='str'),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ['state', 'present', ['interval_unit', 'interval_length', 'remote_location']]
+        ]
+    )
+
+    start_date = module.params.get('start_date')
+    start_time = module.params.get('start_time')
+    interval_unit = module.params.get('interval_unit')
+    interval_length = module.params.get('interval_length')
+    remote_location = module.params.get('remote_location')
+    remote_path = module.params.get('remote_path')
+    state = module.params.get('state')
+
+    mso = MSOModule(module)
+    api_path = 'backups/schedule'
+    mso.existing = mso.request(api_path, method='GET')
+
+    if state == 'absent':
+        mso.previous = mso.existing
+        if module.check_mode:
+            mso.existing = {}
+        else:
+            mso.existing = mso.request(api_path, method='DELETE')
+
+    elif state == 'present':
+        mso.previous = mso.existing
+
+        remote_location_info = mso.lookup_remote_location(remote_location)
+
+        # If no time has been provided default to midnight "00:00:00".
+        try:
+            start_time = time.fromisoformat(start_time) if start_time else time()
+        except Exception as e:
+            module.fail_json(msg="Failed to parse time format 'HH:MM:SS' %s', %s" % (start_time, e))
```

Comment:
why is there a single quote only after the first %s ?
Would probably use "Failed to parse %s with time format 'HH:MM:SS', %s"

Reply:
typo, have changed